### ISSUE TITLE
Branching and release strategy

### ISSUE THREAD

**Issue by ppshobi** [meilisearch/meilisearch-php#17]
Hi, as of now, we are directly merging commits on to the `master` branch, which at this point I assume is fine. But I would like to propose a better model, where the contributors check out from master to either

- a `feature/feature-*` branch 
- a `hotfix/hotfix-*` branch

And should create a PR against the **`develop`** branch instead of `master`. `hotfix/hotfix-*` branches may be issued against the `master` branch. once we have enough features/changes in `develop` branch the maintainer could merge `develop` on to `master` (maybe cleanup the commit history by squashing nasty commits during the process) and issue a release by changing the version number. Security patches maybe directly make it to the master directly.

so the master branch stays stable, developers will have a clean slate to start with which in turn will help in reducing the conflicts as well. Also, we have a streamlined releasing process and each version has something meaningful in it.

If we add this in the Readme.md other devs could easily follow this as well.

**Comment by curquiza** [comment]
Hello @ppshobi!
Thanks for your really interesting suggestion.

To be honest, we (all the team and the CTO) thought about this workflow when we started all our projects. But we decided to keep a more simple workflow for the moment: indeed, our projects are not stable enough, we are doing too many changes every week.
Thus, even the MeiliSearch repository does not use a `develop` branch for the moment 😉

But, I agree with you, this kind of workflow is more professional, and we planned to integrate it when we will be ready! At this moment, can I come back to you if we need help? I am particularly interested in how to explain this workflow in a simple way on the README 🙂

PS: I configured the pushes to Packagist: a WebHook is called on each release I create, and not on each change on `master`. Thus, the package on Packagist stays stable for the users 🙂
